jupyter_kernel_mgmt/subproc/manager.py

```diff
@@ -41,30 +40,17 @@ def __init__(self, popen, files_to_cleanup=None, win_interrupt_evt=None):
         self.win_interrupt_evt = win_interrupt_evt
         self.log = get_app_logger()
         self.kernel_id = str(uuid.uuid4())
+        self._exit_future = asyncio.ensure_future(self.kernel.wait())
 
-    def wait(self, timeout):
+    async def wait(self, timeout):
```

Comment:
If we're going fully async, I think we'll want to remove timeout arguments, and let application code use `asyncio.wait_for()` if it wants to impose a timeout.

Reply:
(We can leave that for a separate PR if you like, just mentioning it as it occurs to me)

Reply:
Good point.  I suspect this will require test changes as well.

Reply:
Hmm - why not just make `timeout` optional?  What is the rationale for putting this one the user?

Reply:
Consistency - with async, you don't need to have a timeout parameter being passed through every method to an underlying method, you don't have to remember if this one takes a number of seconds, or milliseconds, or a timedelta object, etc. You have one way of applying a timeout to everything you can wait for.

Reply:
I see - that makes sense.  Thanks.

Reply:
hmm, I think we still need to wrap `self.kernel.wait()` in `asyncio.wait_for()` - otherwise we get "task is pending" warnings.

Reply:
What do those warnings say? I haven't done a massive amount with asyncio.

Reply:
Sorry - red herring spurred on by the pycharm warning - so I had removed the await prior to wait - which should not be the case.  However, here's an example of the warning...
```
Task was destroyed but it is pending!
task: <Task pending coro=<Process.wait() running at /opt/anaconda3/envs/kernel-mgmt-dev/lib/python3.6/asyncio/subprocess.py:129> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x1077d0d68>()]>>
```

Having restored the `await`, here's what I'm showing for the wait() method.  Not thrilled with catching `Exception` but I'm not sure what choice there is...

```python
    async def wait(self):
        """"""
        try:
            await self.kernel.wait()
            return False
        except Exception:
            return True
```

Btw, doesn't look like any tests are affected since nothing calls wait directly (or with a timeout). :smile:

Reply:
Maybe that `except` was partly to catch the timeout exception. Let's try getting rid of it.

Reply:
Yeah, I've gone ahead and removed the return value as well since only returning False isn't necessary.